port/common/omrargscan.c

```diff
@@ -0,0 +1,411 @@
+/*******************************************************************************
+ * Copyright (c) 1991, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <math.h>
+#include "omr.h"
+#include "omrargscan.h"
+#include "omrport.h"
+#include "omrutil.h"
+
+/* Returns the trimmed input string, removing leading and trailing whitespace.
+ * Returned string is malloc-ed and must be freed.
+ * @param portLibrary The portLibrary to use
+ * @param input The buffer to trim.
+ */
+char * trim(OMRPortLibrary *portLibrary, char * input)
+{
+	BOOLEAN whitespace = TRUE;
+	uintptr_t index;
+	char * buf = (char *)portLibrary->mem_allocate_memory(portLibrary,
+							      strlen(input)+1,
+							      OMR_GET_CALLSITE(),
+							      OMRMEM_CATEGORY_VM);
+
+	while (whitespace == TRUE) {
+		char c = input[0];
+		switch (c) {
+		case '\t':
+		case '\n':
+		case '\r':
+		case ' ':
+			input++;
+			break;
+
+		default:
+			whitespace = FALSE;
+		}
+	}
+
+	strcpy(buf, input);
+	index = strlen(buf) - 1;
+	while (whitespace == TRUE) {
+		char c = buf[index];
+		switch (c) {
+		case '\t':
+		case '\n':
+		case '\r':
+		case ' ':
+			buf[index] = 0;
+			index--;
+			break;
+
+		default:
+			whitespace = TRUE;
+		}
+	}
+
+	return buf;
+}
+
+/* Answer string up to specified delimiter or end of line */
```

Comment:
Please add doxygen comment describing what the function does, what are the input parameters and return value. Similarly for other functions in this file.